src/Raven.Server/Indexing/LuceneVoronDirectory.cs

```diff
@@ -230,9 +230,5 @@ public void ResetAllocations()
         {
             _indexOutputFilesSummary.Reset();
         }
-
-        protected override void Dispose(bool disposing)
```

Comment:
Did it cause any problems?

Reply:
It was hiding the Dispose of `Directory`.
https://github.com/ravendb/lucenenet/blob/34bc333c964b47e4c8069e726ba1d316d07eb664/src/Lucene.Net/Store/Directory.cs#L163

Since we didn't have a reference to `LuceneVoronDirectory`, the finalizer would dispose of all the `ArrayHolder` instances.

Reply:
Nice catch